integration-tests/bats/events.bats

```diff
@@ -0,0 +1,173 @@
+
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SHOW CREATE EVENT insert9;"
+    [ $status -eq 0 ]
+    [[ $output =~ "ON COMPLETION PRESERVE DISABLE" ]] || false
+}
+
+@test "events: recurring event with STARTS and ENDS defined" {
+    cd repo1
+    dolt sql -q "CREATE TABLE totals (id int PRIMARY KEY AUTO_INCREMENT, int_col int)"
+
+    start_sql_server
+    dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "CREATE EVENT insert1 ON SCHEDULE EVERY 2 SECOND STARTS CURRENT_TIMESTAMP + INTERVAL 2 SECOND ENDS CURRENT_TIMESTAMP + INTERVAL 5 SECOND DO INSERT INTO totals (int_col) VALUES (1);"
+    sleep 10
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SELECT COUNT(*) FROM totals;"
+    [ $status -eq 0 ]
+    [[ $output =~ "| 2        |" ]] || false
+
+    # should be dropped
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SELECT COUNT(*) FROM information_schema.events;"
+    [ $status -eq 0 ]
+    [[ $output =~ "| 0        |" ]] || false
+}
+
+@test "events: recurring event with ENDS defined" {
+    cd repo1
+    dolt sql -q "CREATE TABLE totals (id int PRIMARY KEY AUTO_INCREMENT, int_col int)"
+
+    start_sql_server
+    dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "CREATE EVENT insert1 ON SCHEDULE EVERY 2 SECOND ENDS CURRENT_TIMESTAMP + INTERVAL 3 SECOND ON COMPLETION PRESERVE DO INSERT INTO totals (int_col) VALUES (1); SELECT SLEEP(5);"
+    sleep 2
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SELECT COUNT(*) FROM totals;"
+    [ $status -eq 0 ]
+    [[ $output =~ "| 2        |" ]] || false
+
+    # should be disabled
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SELECT * FROM information_schema.events;"
+    [ $status -eq 0 ]
+    [[ $output =~ "DISABLED" ]] || false
+}
+
+@test "events: checking out a branch should disable all events leaving the working set dirty" {
+    cd repo1
+    dolt sql -q "CREATE TABLE totals (id int PRIMARY KEY AUTO_INCREMENT, int_col int)"
+    dolt add .
+    dolt commit -m "create table"
+
+    start_sql_server
+    dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "CREATE EVENT insert1 ON SCHEDULE EVERY 2 SECOND ENDS CURRENT_TIMESTAMP + INTERVAL 3 SECOND ON COMPLETION PRESERVE DO INSERT INTO totals (int_col) VALUES (1); SELECT SLEEP(5);"
+    dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "CALL DOLT_COMMIT('-am','commit with an event')"
+
+    dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "CALL DOLT_CHECKOUT('-b','newbranch')"
+    # should be disabled
+    run dolt sql-client -P $PORT -u dolt --use-db 'repo1' -q "SELECT * FROM information_schema.events;"
+    [ $status -eq 0 ]
+    [[ $output =~ "DISABLED" ]] || false
+}
+
+@test "events: events on default branch still runs after switching to non default branch" {
+    cd repo1
+    dolt sql -q "CREATE TABLE totals (id int PRIMARY KEY AUTO_INCREMENT, int_col int)"
+    dolt add .
+    dolt commit -m "create table"
```

Comment:
Probably worth putting this into the setup code so that it's consistent for all tests, since they all need a test table to work with.